src/JSONNestedNode.js

```diff
@@ -131,6 +140,30 @@ export default class JSONNestedNode extends React.Component {
     );
     const stylingArgs = [keyPath, nodeType, expanded, expandable];
 
+    const handleClick = () => {
+      const newState = !expanded;
+
+      if (this.props.onNodeExpansionChanging) {
+        this.props.onNodeExpansionChanging(
+          this.props.keyPath,
+          this.props.data,
+          this.props.level,
+          newState);
+      }
+
+      this.setState(
+        { expanded: newState },
+        () => {
+          if (this.props.onNodeExpansionChanged) {
+            this.props.onNodeExpansionChanged(
+              this.props.keyPath,
+              this.props.data,
+              this.props.level,
+              this.state.expanded);
+          }
+        });
+    };
+
```

Comment:
After https://github.com/alexkuz/react-json-tree/pull/61 it's better to move this back to class property.